File: gateway-api/internal/handlers/dto/session.go
package dto

type SessionResponse struct {
	ID           string               `json:"id"`
	Name         string               `json:"name"`
	Total        float64              `json:"total"`
	IsActive     bool                 `json:"isActive"`
	Members      []SessionMember      `json:"members"`
	Transactions []SessionTransaction `json:"transactions"`
}

type SessionMember struct {
	ID                 string             `json:"id"`
	Name               string             `json:"name"`
	Username           string             `json:"username"`
	TransactionSummary TransactionSummary `json:"transactionSummary"`
}

type SessionTransactionMember struct {
	ID       string  `json:"userId"`
	Name     string  `json:"name"`
	Username string  `json:"username"`
	Amount   float64 `json:"amount"`
}

type SessionTransaction struct {
	ID        string                     `json:"id"`
	CreatorID string                     `json:"creatorId"`
	Total     float64                    `json:"total"`
	CreatedAt string                     `json:"createdAt"`
	Members   []SessionTransactionMember `json:"members"`
}

type TransactionSummary struct {
	Credit float64 `json:"credit"`
	Debit  float64 `json:"debit"`
}

File: webui/src/api/types/session.ts
export interface Session {
	id: string;
	name: string;
	isActive: boolean;
	members: SessionMember[];
}

export interface SessionWithTransactions extends Session {
	transactions: SessionTransaction[];
	total: number;
}

export type SessionMember = {
	id: string;
	name: string;
	username: string;
	isCreator: boolean;
	isAdmin: boolean;
	isDeleted: boolean;
	transactionSummary: MemberTransactionSummary;
};

type MemberTransactionSummary = {
	credit: number;
	debit: number;
};

export type SessionTransaction = {
	id: string;
	userId: string;
	total: number;
	createdAt: string;
	lines: SessionTransactionLine[];
};

export type SessionTransactionLine = {
	userId: string;
	name: string;
	username: string;
	amount: number;
};
